packages/esbuild/lib/index.js

```diff
@@ -157,44 +158,60 @@ export function esbuild(options) {
  *   ESBuild message.
  */
 function vfileMessageToEsbuild(state, message) {
+  /** @type {Location} */
+  let location = {
+    column: 0,
+    file: state.path,
+    length: 0,
+    line: 0,
+    lineText: '',
+    namespace: 'file',
+    suggestion: '',
+  }
+
   const place = message.place
   const start = place ? ('start' in place ? place.start : place) : undefined
-  const end = place && 'end' in place ? place.end : undefined
-  let length = 0
-  let lineStart = 0
-  let line = 0
-  let column = 0
-
-  if (start && start.offset !== undefined) {
-    line = start.line
-    column = start.column - 1
-    lineStart = start.offset - column
-    length = 1
-
-    if (end && end.offset !== undefined) {
-      length = end.offset - start.offset
+  if (start) {
+    location.column = start.column - 1
+    location.line = start.line
+    location.length = 1
+
+    const end = place && 'end' in place ? place.end : undefined
+    if (end) {
+      if (start.offset !== undefined && end.offset !== undefined) {
+        location.length = end.offset - start.offset;
+      } else if (end.line === start.line) {
+        location.length = end.column - start.column;
+      }
     }
-  }
 
-  eol.lastIndex = lineStart
+    if (start.offset !== undefined) {
+      eol.lastIndex = start.offset
+      const match = eol.exec(state.doc)
+      const lineStart = start.offset - (start.column - 1)
+      const lineEnd = match ? match.index : state.doc.length
+      location.lineText = state.doc.slice(lineStart, lineEnd)
+      location.length = Math.min(location.length, lineEnd - (start.offset || 0))
+    }
+
+    const maxLength = state.doc.length - (start.offset || 0)
+    location.length = Math.min(location.length, maxLength)
+  }
 
-  const match = eol.exec(state.doc)
-  const lineEnd = match ? match.index : state.doc.length
+  /** @type {Error} */
+  var exc = message
+  var text = message.reason
+  while (exc.cause instanceof Error) {
+    exc = exc.cause
+    text = `${text}:\n  ${exc}`
+  }
 
   return {
     detail: message,
     id: '',
-    location: {
-      column,
-      file: state.path,
-      length: Math.min(length, lineEnd),
-      line,
-      lineText: state.doc.slice(lineStart, lineEnd),
-      namespace: 'file',
-      suggestion: ''
-    },
+    location,
     notes: [],
     pluginName: state.name,
-    text: message.reason
+    text,
```

Comment:
Please use the npm scripts in this package: `npm test` is the main one. It uses `npm run format`. That formats things: it removes that `,`

Reply:
(done)